doc/cprover-manual/modeling-shadow-memory.md

```diff
@@ -151,11 +151,94 @@ void func() {
     // to a value of `1`.
     __CPROVER_set_field(&a, "shadow", 1);
     // Retrieve the value of the field named "shadow" associated with the address
-    // of the object `a`. 
+    // of the object `a`.
     shadow_x = __CPROVER_get_field(&a, "shadow");
     __CPROVER_assert(shadow_x == 1, "expected success: field set to a value of 1");
 }
 ```
 
 Note that setting the value of a local variable from a global SSM field or the
 opposite will produce no effect (and it **will not** fail).
+
+### Working with Compound Type Objects
+
+When using SSM on compound type pointers (e.g. `struct`) the value used for the
```

Comment:
Does this description of shadow memory as it applies to struct types also apply to unions and arrays?